### PR TITLE
Use the new labels mechanism in docs guide reorganization

### PR DESCRIPTION
**Pull request by Eric-Arellano**
This also improves the patterns index pages:

* adds missing metadata
* uses `dedent()` so the code is easier to modify
* Adds an h2 `Guides`

### CODE REVIEW

File: scripts/patterns-reorg/page_content.py [hunk 1]
```diff
@@ -14,69 +14,126 @@
...
+---
+
+# Optimize for target hardware
+
+In the "optimize for target hardware" step of a Qiskit pattern, you take the abstract circuits
+(or operators) produced from the map step and perform a series of optimizations on them. This
+may include mapping the route and layout of the circuit to physical qubit hardware, converting
+to basis gates of the hardware, and reducing the number of operations, all designed to optimize
+the likelihood of success in the later execute step. At this point you may also wish to debug
+your circuits with a simulator before executiong on real hardware in the next step.
```
Comment (arnaucasau) [inline scripts/patterns-reorg/page_content.py:84]:
```suggestion
your circuits with a simulator before executing on real hardware in the next step.
```

File: scripts/patterns-reorg/page_content.py [hunk 2]
```diff
@@ -14,69 +14,126 @@
...
+    intro = dedent(
+        """\
+---
+title: Execute on hardware
+description: TODO - 50+ characters long to ignore metadata check for now!
+---
+
+# Execute on hardware
+
+The "execute on hardware" step of a Qiskit pattern involes running you circuits on hardware
```
Comment (arnaucasau) [inline scripts/patterns-reorg/page_content.py:107]:
```suggestion
The "execute on hardware" step of a Qiskit pattern involves running your circuits on hardware
```

File: scripts/patterns-reorg/page_content.py [hunk 3]
```diff
@@ -14,69 +14,126 @@
...
+
+## Guides
+"""
     )
+    return f"{intro}{index}\n"
 
 
 def postprocess_index_content(index: str) -> str:
-    return (
-        "# Post-process results\n\n"
```
Comment (arnaucasau) [inline scripts/patterns-reorg/page_content.py:37]: I see that all the other sections preserve the title except this one. Is this intentional?

**Review by arnaucasau — COMMENTED** [review]
Thanks Eric!

File: scripts/patterns-reorg/page_content.py (hunk 3 above)
Comment (Eric-Arellano) [reply scripts/patterns-reorg/page_content.py:37]: Oh, oops. Good catch!

File: scripts/patterns-reorg/page_content.py [hunk 4]
```diff
@@ -14,69 +14,126 @@
...
-        "unphysical observables.\n\n"
-        "There are a number of software tools that could be used for this step, you can check out the "
-        f"following resources:\n\n{index}\n"
+    intro = dedent(
+        """\
+---
+title: Post-process results
+description: TODO - 50+ characters long to ignore metadata check for now!
+---
+
```
Comment (Eric-Arellano) [inline scripts/patterns-reorg/page_content.py:55]:
```suggestion

# Post-process results

```

**Review by arnaucasau — APPROVED** [review]

**Pull request merged**